Core/src/Scene.cpp

```diff
@@ -65,6 +66,17 @@ void Scene::SpawnSceneDefinition()
     ScriptComponent* orb = dynamic_cast<ScriptComponent*>(cameraEnt->getComponent(ComponentType::Script));
     OrbitScript* orbScript = orb->GetScript<OrbitScript>();
 
+    // Audio setup
+    AudioManager* am = AudioManager::GetInstance();

```

Comment:
We violate ECS by referring to a system in this way inside scene. Remember that AudioManager is a system, we don't get a system anywhere else here in Scene.

This probably means there's room to split AudioManager into its system behaviours, and some other functionality that youre using in this way...

Reply:
This is where i'm at an impasse.

The trouble is that we need audio to be updated with the update loops of the engine, but also everything to do with sound *needs* to be able to access the system to perform any actions. It's how fmod is built.

To play any sound, you must call the fmod system member function PlaySound. In a world where this wasn't a system that would be fine, but the system itself also needs to be updated with the rest of the things, like physics.

Reply:
See top level review comment

```
RE: Comments in AudioComponent, AudioListener and AudioManager and its relationship to Scene -
In summary you should probably split AudioManager (which is actually a system) into an actual FModManager.h class (for which a singleton would be perfectly acceptable and do all the variable associations in your Listeners and audio components) and an Audio.h, Audio.cpp ECS SYSTEM. Which behaves like the rest of the systems without interdependencies.

FModManager::getInstance()->GetFMODSystem()-><do something>

Now we dont have to store a system as a variable, we dont have to pass it in as a param, we dont have to make order dependent code
```